samples/philosophers/sample.kernel.philosopher.stacks fails on xtensa

`./scripts/twister -s samples/philosophers/sample.kernel.philosopher.stacks -a xtensa -l`

Gets:

```
ERROR   - qemu_xtensa               samples/philosophers/sample.kernel.philosopher.stacks FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/qemu_xtensa/samples/philosophers/sample.kernel.philosopher.stacks/build.log
INFO    - Total complete:    1/   3  33%  skipped:    8, failed:    1
ERROR   - intel_s1000_crb           samples/philosophers/sample.kernel.philosopher.stacks FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/twister-out/intel_s1000_crb/samples/philosophers/sample.kernel.philosopher.stacks/build.log
```

Log has:
```
/home/galak/git/zephyr/subsys/debug/thread_info.c:87:2: error: #warning Please define THREAD_INFO_OFFSET_T_STACK_PTR for this architecture [-Werror=cpp]
   87 | #warning Please define THREAD_INFO_OFFSET_T_STACK_PTR for this architecture
      |  ^~~~~~~
cc1: all warnings being treated as errors

```

--- a/samples/philosophers/src/phil_obj_abstract.h
+++ b/samples/philosophers/src/phil_obj_abstract.h
@@ -81,12 +81,12 @@
 			uint32_t stack_mem[1];
 		} fork_obj_t;
 		#define fork_init(x) do { \
-			k_stack_init(x, (uint32_t *)((x) + 1), 1); \
+			k_stack_init(x, (void *)((x) + 1), 1); \
 			k_stack_push(x, MAGIC); \
 		} while ((0))
 	#endif
 	#define take(x) do { \
-		uint32_t data; k_stack_pop(x, &data, K_FOREVER); \
+		uint32_t data; k_stack_pop(x, (void *)&data, K_FOREVER); \
 		__ASSERT(data == MAGIC, "data was %x\n", data); \
 	} while ((0))
 	#define drop(x) k_stack_push(x, MAGIC)

--- a/subsys/debug/thread_info.c
+++ b/subsys/debug/thread_info.c
@@ -80,6 +80,13 @@ size_t _kernel_thread_info_offsets[] = {
 #elif defined(CONFIG_ARCH_POSIX)
 	[THREAD_INFO_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
 						callee_saved.thread_status),
+#elif defined(CONFIG_XTENSA)
+	/* Xtensa does not store stack pointers inside thread objects.
+	 * The registers are saved in thread stack where there is
+	 * no fixed location for this to work. So mark this as
+	 * unimplemented to avoid the #warning below.
+	 */
+	[THREAD_INFO_OFFSET_T_STACK_PTR] = THREAD_INFO_UNIMPLEMENTED,
 #else
 	/* Use a special value so that OpenOCD knows that obtaining the stack
 	 * pointer is not possible on this particular architecture.

